pkg/controller/agent/fleet.go

```diff
@@ -276,22 +308,33 @@ func reconcileEnrollmentToken(ctx context.Context, agent agentv1alpha1.Agent, cl
 		}
 	}
 
-CREATE:
-	key, err := api.createEnrollmentAPIKey(ctx, policyID)
+FindOrCreate:
+	// try to re-use an existing key
+	newkey := false
+	key, err := api.findEnrollmentAPIKey(ctx, policyID)
 	if err != nil {
-		return EnrollmentAPIKey{}, err
+		ulog.FromContext(ctx).Info("Could not find existing Fleet enrollment API keys, creating new one", "error", err.Error())
```

Comment:
Should we only create a new one if the error is `404`?

Reply:
I think we also want to create a new one if there is a bunch of old tokens but they are either all for other policies or have been disabled by the user.

Reply:
What if `findEnrollmentAPIKey` fails with a timeout for example? In that case we don't know if there is an existing and valid token, we may create a new one while it is not necessary.

Reply:
Good point. We can check for the specific error as you suggested in your updated comment above.